src/Rules/UnusedPublicClassMethodRule.php

```diff
@@ -68,6 +72,10 @@ public function processNode(Node $node, Scope $scope): array
 
         foreach ($publicClassMethodCollector as $filePath => $declarations) {
```

Comment:
It would be more reliable to filter the public method in collector itself. So here we only validate methods we are interested in.

The `$publicClassMethodCollector` should contain only `Class_` methods.

Reply:
good catch, indeed. fixed.

Reply:
What about hooking directly to `Class_` node? That would solve this backtracking magic. Also would skip the interfaces and enums.

Reply:
In:

```diff
public function getNodeType(): string
    {
-        return ClassMethod::class;
+       return Class_::class;
    }
```

Reply:
I tried but I think its not that easy. I think we should separate it into a new PR

Reply:
Shall I handle it?

Reply:
sure if you like